modules/apps/frontend-css/frontend-css-cadmin-web/src/main/java/com/liferay/frontend/css/cadmin/web/internal/servlet/taglib/CadminTopHeadDynamicInclude.java

```diff
@@ -0,0 +1,108 @@
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.css.cadmin.web.internal.servlet.taglib;
+
+import com.liferay.portal.kernel.security.permission.PermissionChecker;
+import com.liferay.portal.kernel.security.permission.PermissionThreadLocal;
+import com.liferay.portal.kernel.servlet.taglib.BaseDynamicInclude;
+import com.liferay.portal.kernel.servlet.taglib.DynamicInclude;
+import com.liferay.portal.url.builder.AbsolutePortalURLBuilder;
+import com.liferay.portal.url.builder.AbsolutePortalURLBuilderFactory;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Julien Castelain
+ */
+@Component(immediate = true, service = DynamicInclude.class)
+public class CadminTopHeadDynamicInclude extends BaseDynamicInclude {
+
+	@Override
+	public void include(
+			HttpServletRequest httpServletRequest,
+			HttpServletResponse httpServletResponse, String key)
+		throws IOException {
+
+		if (!isSignedIn()) {
+			return;
+		}
+
+		PrintWriter printWriter = httpServletResponse.getWriter();
+
+		AbsolutePortalURLBuilder absolutePortalURLBuilder =
+			_absolutePortalURLBuilderFactory.getAbsolutePortalURLBuilder(
+				httpServletRequest);
+
+		for (String fileName : _FILE_NAMES) {
+			printWriter.print("<link data-senna-track=\"temporary\" href=\"");
+
+			printWriter.print(
+				absolutePortalURLBuilder.forModule(
+					_bundleContext.getBundle(), fileName
+				).build());
+
+			printWriter.println(
+				"\" id=\"liferayCadminCSS\" rel=\"stylesheet\"");
+			printWriter.println(" type=\"text/css\" />");
+		}
+	}
+
+	@Override
+	public void register(DynamicIncludeRegistry dynamicIncludeRegistry) {
+		dynamicIncludeRegistry.register("/html/common/themes/top_head.jsp#pre");
+	}
+
+	@Activate
+	@Modified
+	protected void activate(
+			BundleContext bundleContext, ComponentContext componentContext,
+			Map<String, Object> properties)
+		throws Exception {
+
+		_bundleContext = bundleContext;
+	}
+
+	protected boolean isSignedIn() {
+		PermissionChecker permissionChecker =
+			PermissionThreadLocal.getPermissionChecker();
+
+		if ((permissionChecker == null) || !permissionChecker.isSignedIn()) {
+			return false;
+		}
+
+		return true;
+	}
+
+	private static final String[] _FILE_NAMES = {"main.css"};
```

Comment:
If it will be just one file maybe we can change this to a scalar `String` and get rid of the `for` loop at lines 61-72... :thinking: